src/layouts/sidebar-sidecar/index.tsx

```diff
@@ -7,31 +7,48 @@ import Footer from 'components/footer'
 import Sidebar from 'components/sidebar'
 import { SidebarSidecarLayoutProps } from './types'
 import s from './sidebar-sidecar-layout.module.css'
+import {
+  SidebarNavDataProvider,
+  useSidebarNavData,
+} from './contexts/sidebar-nav-data'
 
-const SidebarSidecarLayout = ({
+const SidebarSidecarLayout = (props: SidebarSidecarLayoutProps) => {
+  const navDataLevels = props.sidebarNavDataLevels || [props.sidebarProps]
+  return (
+    <SidebarNavDataProvider navDataLevels={navDataLevels}>
+      <SidebarSidecarLayoutContent
+        {...props}
+        sidebarNavDataLevels={navDataLevels}
+      />
+    </SidebarNavDataProvider>
+  )
+}
+
+const SidebarSidecarLayoutContent = ({
   breadcrumbLinks,
   children,
   githubFileUrl,
   headings,
   openConsentManager,
-  sidebarProps,
   sidebarSlot,
   sidecarSlot,
+  sidebarNavDataLevels,
 }: SidebarSidecarLayoutProps) => {
+  const { currentLevel } = useSidebarNavData()
+  const sidebarProps = sidebarNavDataLevels[currentLevel]
+
+  /**
+   * @TODO the docs Sidebar can have props spread onto it but not all uses of
+   * sidebarSlot allow props spreading. The spreading may also become
+   * unnecessary once there is a Context that helps manage the current sidebar
+   * level and how to change it.
+   */
```

Comment:
📌 TODO: test to double-check if this comment is still relevant or not (solution has evolved since I added it)

Reply:
🗒️ Update: branched off of this one to start looking at how this'll all work for Learn content, and figured out a way to make it work. This comment will go away in the next PR (#351). 😊